### ISSUE TITLE
Inconsistent binding with "with" helper

### ISSUE THREAD

**Issue by whitecolor** [canjs/canjs#2129]
2.3.5 latest:

```HTML
<script id="app-template" type="text/stache">
<comp-one>
  {{#with someMap}}
  <comp-two {attr-a}="attrA" {attr-b}="attrB" {^attr-c}="attrC"></comp-two>
  {{/with}}
</comp-one>
</script>
```
```javascript
console.clear()
can.Component.extend({
  tag: "comp-one",
  template: can.stache('<content/>'),
  viewModel: {
    attrC: "",
    attrB: "ATTRB",
    someMap: {attrC: ""}
  }
});

can.Component.extend({
  tag: "comp-two",
  viewModel: {
    attrC: "ATTRC"
  } 
});


var outer = new can.Map({
  attrA: "ATTRA"
})

$("body").prepend(
  can.view("app-template",outer) 
);

//Sould comp-on scope attrC be 'ATTRC'?
//Sould comp-on scope someMap.attrC be 'ATTRC'?

console.log('comp-one', $("comp-one").viewModel().serialize() )
console.log('comp-two', $("comp-two").viewModel().serialize() )
```

http://jsbin.com/fejabegore/1/edit?html,js,console,output

`{^attr-c}="attrC"` seem to have place `attrC` on the parent scope, in example it doesn't place it nor on viewModel, nor on `someMap` that is set using `{{#with someMap}}`

2.3.4 gives an error with such example:
http://jsbin.com/yalutefuye/1/edit?html,js,console,output

I believe if we using "with" in this case attrC should be placed on viewModel, but I'm not sure.

**Comment by whitecolor** [comment]
this is related to https://github.com/canjs/canjs/issues/2122 and I think this as well https://github.com/canjs/canjs/issues/2092

**Comment by justinbmeyer** [comment]
Yeah, this should be fixed by #2139.  Please re-open if you continue to find this.
